test/pinnedLinks.js

```diff
@@ -0,0 +1,148 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+/*globals gPinnedLinks, gUserDatabase */
+
+describe("Pinned Links API", () => {
+  "use strict";
+
+  var firstLink = {url: "http://example0.com/", title: "site#0"};
+  var secondLink = {url: "http://example1.com/", title: "site#1"};
+  var directoryLink = {url: "http://directorylink.com", title: "DirectoryLink", type: "affiliate"};
+
+  var linksToSet = JSON.stringify([
+    firstLink, secondLink,
+    {url: "http://example2.com/", title: "site#2"},
+    {url: "http://example3.com/", title: "site#3"},
+    {url: "http://example4.com/", title: "site#4"}
+  ]);
+
+  afterEach(function(done) {
+    // Clear the database and cached links.
+    gPinnedLinks.resetCache();
+    gUserDatabase.init().then(pinnedLinks => {
+      gPinnedLinks.setLinks(pinnedLinks);
+      gUserDatabase.save("prefs", "pinnedLinks", []).then(() => {
+        gUserDatabase.close();
+        done();
+      });
+    });
+  });
+
+  it("Links should be empty initially", () => {
```

Comment:
`it()` methods should begin with "should". If you are describing some particular aspect of the api, please wrap the test in `describe("Links", function(){})`.